### ISSUE TITLE
Way to use this

### ISSUE THREAD

**Issue by BDhackers009** [bdureau/ESPLoader#1]
Sir, is there any tutorial or article about using this? I am trying to flash a firmware to my esp32 but seems like there are no such applications available. I found your GitHub repo while researching please help me if possible.

**Comment by bdureau** [comment]
Hello
You will need Eclipse. The test program is [testESPLoader.java](https://github.com/bdureau/ESPLoader/blob/main/src/ESPLoader/testESPLoader.java)
This is a simple proof of concept to demonstrate my port of ESPLoader to Java

**Comment by BDhackers009** [comment]
Oh. Which means we cannot use this program directly on android unless we make an apk out of it?

is there any POC video which demonstrates this?. Thanks for your reply sir.

**Comment by bdureau** [comment]
I have another piece of code that runs on Android

**Comment by bdureau** [comment]
check the following https://github.com/bdureau/BearConsole2/blob/master/app/src/main/java/com/altimeter/bdureau/bearconsole/Flash/CommandInterfaceESP32.java

**Comment by bdureau** [comment]
My BearConsole application is using it to flash an ESP32 firmware

**Comment by BDhackers009** [comment]
> check the following https://github.com/bdureau/BearConsole2/blob/master/app/src/main/java/com/altimeter/bdureau/bearconsole/Flash/CommandInterfaceESP32.java

seems like a good code. Though I don't understand java like you. Could you please tell me how Do i use this piece of code? I mean Do i need to build this BearConsole app ? or i can use it any other way? Also I couldn't find the firmware file input variable or something like that. is the code do this Interactively?

Please sir any information related to this will help me a lot.

**Comment by BDhackers009** [comment]
![Screenshot_20240325_191652_Chrome](https://github.com/bdureau/ESPLoader/assets/67186139/6529f1f3-11fd-4426-8b75-51fe596f0c67)


If i Change the firmware.bin and bootloader bin and other bin files which are present here then that should work on my normal esp32 right? or i can just copy and paste codes to create another esp32 options in the apps firmware section?

**Comment by bdureau** [comment]
yes it should work. Are you trying to build your own Android app to flash a firmware?

**Comment by BDhackers009** [comment]
> yes it should work. Are you trying to build your own Android app to flash a firmware?

I am  just trying to do something so that i can flash my esp32 devices via my android. :( . But there isn't any source that can do that except your app. Its flashing the firmware correctly ig. I have to change the files and offsets in the flashfirmware.java ig.

**Comment by BDhackers009** [comment]
> > yes it should work. Are you trying to build your own Android app to flash a firmware?
> 
> I am just trying to do something so that i can flash my esp32 devices via my android. :( . But there isn't any source that can do that except your app. Its flashing the firmware correctly ig. I have to change the files and offsets in the flashfirmware.java ig.

Trying to flash the firmware with these files and offsets. https://raw.githubusercontent.com/Fr4nkFletcher/ESP32-Marauder-Cheap-Yellow-Display/master/web/manifest_cyd2usbnogps.json

**Comment by yusufcirak** [comment]
Hello.
i need to update chips like esp32 in my flutter project. i found your source on the internet. but i was not successful.

currently there is serial communication in my project. i tried with the loader you shared but i was not fully successful. it worked 1 time in total and it took 10 minutes

**Comment by yusufcirak** [comment]
[uart.zip](https://github.com/user-attachments/files/18565853/uart.zip)         Can you go through it for me and tell me where I'm missing something?
